src/test/java/uk/gov/hmcts/ccd/data/caseaccess/AMCaseUserRepositoryComponentTest.java

```diff
@@ -0,0 +1,123 @@
+import static uk.gov.hmcts.reform.amlib.enums.SecurityClassification.PUBLIC;
+
+@Transactional
+public class AMCaseUserRepositoryComponentTest extends BaseTest {
+
+    private static final String JURISDICTION_ID = "JURISDICTION";
+    private static final String CASE_TYPE_ID = "CASE_TYPE";
+    private static final String CASE_REFERENCE = "1234123412341234";
+    private static final Long CASE_ID = 1L;
+    private static final Long CASE_ID_GRANTED = 2L;
+    private static final Long CASE_ID_3 = 3L;
+    private static final String USER_ID = "89000";
+    private static final String USER_ID_GRANTED = "89001";
+    private static final String CASE_ROLE = "[DEFENDANT]";
+    private static final String CASE_ROLE_SOLICITOR = "[SOLICITOR]";
+    private static final String CASE_ROLE_CREATOR = "[CREATOR]";
+
+    @Autowired
+    private AMCaseUserRepository repository;
+
+    private DefaultRoleSetupImportService defaultRoleSetupImportService;
+
+    private AccessManagementQueryHelper accessManagementQueryHelper;
+
+    @Before
+    public void setUp() throws IOException {
+        DataSource dataSource = AccessManagementQueryHelper.amDataSource();
+        repository.accessManagementService = new AccessManagementService(dataSource);
+        defaultRoleSetupImportService = new DefaultRoleSetupImportService(dataSource);
+        accessManagementQueryHelper = new AccessManagementQueryHelper(dataSource);
+
+        defaultRoleSetupImportService.addService(JURISDICTION_ID);
+        defaultRoleSetupImportService.addRole(CASE_ROLE, IDAM, PUBLIC, ROLE_BASED);
+        defaultRoleSetupImportService.addRole(CASE_ROLE_SOLICITOR, IDAM, PUBLIC, ROLE_BASED);
+        defaultRoleSetupImportService.addRole(CASE_ROLE_CREATOR, IDAM, PUBLIC, ROLE_BASED);
+
+        ResourceDefinition resourceDefinition =
+            //TODO: What should be the resourceType and resourceName.
+            //To be clarified with Mutlu/Shashank again.//resource name: CMC, FPL
+            new ResourceDefinition(JURISDICTION_ID, "case", CASE_REFERENCE);
+        defaultRoleSetupImportService.addResourceDefinition(resourceDefinition);
+    }
+
+    @After
+    public void tearDown() {
+        accessManagementQueryHelper.deleteAllFromAccessManagementTables();
+    }
+
+    @Test
+    public void shouldGrantAccessAsCustomCaseRole() {
+        repository.grantAccess(JURISDICTION_ID, CASE_REFERENCE, CASE_ID, USER_ID, CASE_ROLE);
+        Integer records = accessManagementQueryHelper.findExplicitAccessPermissions(JURISDICTION_ID);
+
+        assertThat(records, equalTo(1));
+    }
+
+    @Test
+    public void shouldRevokeAccessAsCustomCaseRole() {
+        repository.grantAccess(JURISDICTION_ID, CASE_REFERENCE, CASE_ID, USER_ID, CASE_ROLE);
+        repository.revokeAccess(JURISDICTION_ID, CASE_REFERENCE, CASE_ID, USER_ID, CASE_ROLE);
+
+        Integer records = accessManagementQueryHelper.findExplicitAccessPermissions(JURISDICTION_ID);
```

Comment:
This doesn't necessarily verify the correct granting/revoking achieved on the previous line.

Reply:
@MSancaktutar I believe the better way is to verify that the count of records after granting access should be 1. And in next step, revoke the access and count again. Please let me know if this looks good to you.

Reply:
Yes please make sure you verify the record number is 1 after granting.